### ISSUE TITLE
Add support for profiling Gradle syncs with IntelliJ 2023.3+ 

### ISSUE THREAD

**Issue by asodja** [gradle/gradle-profiler#546]
We support IntelliJ sync profiling (it's a bit undocumented but it works).

With IntelliJ 2023.3 Android plugin is not embedded anymore in IntelliJ distribution. Since we use Android plugin APIs to do a sync that means that we don't support IntelliJ 2023.3 at the moment. So we have to use a different API for IntelliJ. 

That should not be the case for Android Studio, since Android plugin is embedded there.   

I believe the problem is using this API
```
ProjectSystemUtil.getSyncManager(project).syncProject()
```
[AndroidStudioSystemHelper.java#L61](https://github.com/gradle/gradle-profiler/blob/72aa9f458a21bce1553e2dca185fc82688d2bc21/subprojects/studio-plugin/src/main/java/org/gradle/profiler/studio/plugin/system/AndroidStudioSystemHelper.java#L61), that is only available in AS.

Instead, we could use for IntelliJ something like:
```
ImportSpec importSpec = new ImportSpecBuilder(project, GradleConstants.SYSTEM_ID)
    .use(ProgressExecutionMode.NO_PROGRESS_SYNC) // Or maybe ProgressExecutionMode.MODAL_SYNC
    .callback(new ExternalProjectRefreshCallback() {
        // Implement callback onSuccess/onFailure methods
    })
    .build();
ExternalSystemUtil.refreshProject(project.getBasePath(), importSpec);
```

**Comment by autonomousapps** [comment]
I would actually like a way to tell gradle-profiler to install specific plugins prior to running, and I believe that's related to this issue. I can actually install the Android plugin in my IntelliJ environment, but because the version of the IDE launched by g-p is "sandboxed," it doesn't respect any of my user config. In other words, I could solve this issue right now by telling g-p to install the Android plugin.

In addition to that though, I'd like to have _other_ plugins installed as well, since there are cases where plugins can be expected to have an impact on sync performance (both improving and harming it).

**Comment by asodja** [comment]
Does it work if you use `--no-studio-sandbox`?

**Comment by autonomousapps** [comment]
> Does it work if you use `--no-studio-sandbox`?

No. I don't think it's intended that way either. Looking at the code, I see the [partial sandbox](https://github.com/gradle/gradle-profiler/blob/341b1544f1d108bda49ff1205eb05a1b0a433b33/src/main/java/org/gradle/profiler/studio/tools/StudioSandboxCreator.java#L129-L131) still includes a separate pluginsDir.

**Comment by asodja** [comment]
The problem why it doesn't work with IntelliJ as far as I remember is using this API
```
ProjectSystemUtil.getSyncManager(project).syncProject()
```
[AndroidStudioSystemHelper.java#L61](https://github.com/gradle/gradle-profiler/blob/72aa9f458a21bce1553e2dca185fc82688d2bc21/subprojects/studio-plugin/src/main/java/org/gradle/profiler/studio/plugin/system/AndroidStudioSystemHelper.java#L61), that is only available in AS.

Instead, we could use for IntelliJ something like:
```
ImportSpec importSpec = new ImportSpecBuilder(project, GradleConstants.SYSTEM_ID)
    .use(ProgressExecutionMode.NO_PROGRESS_SYNC) // Or maybe ProgressExecutionMode.MODAL_SYNC
    .callback(new ExternalProjectRefreshCallback() {
        // Implement callback onSuccess/onFailure methods
    })
    .build();
ExternalSystemUtil.refreshProject(project.getBasePath(), importSpec);
```

**Comment by 6hundreds** [comment]
Implemented by https://github.com/gradle/gradle-profiler/pull/740

**Comment by trobalik-duolingo** [comment]
Is there an open issue for [this comment](https://github.com/gradle/gradle-profiler/issues/546#issuecomment-3046625456)? That is, a way to tell g-p to install certain plugins in its sandboxed version of intellij?

**Comment by 6hundreds** [comment]
@trobalik-duolingo Hey Tony, there is no. Could you file it?

**Comment by autonomousapps** [comment]
https://github.com/gradle/gradle-profiler/issues/822
